core/src/Cuda/Kokkos_Cuda_GraphNodeKernel.hpp

```diff
@@ -34,6 +34,37 @@
 namespace Kokkos {
 namespace Impl {
 
+template <typename Functor>
+struct GraphNodeThenImpl<Kokkos::Cuda, Functor> {
+  Functor m_functor;
+  cudaGraphNode_t m_node;
+
+  // This explicit constructor is not needed by all compilers.
+  explicit GraphNodeThenImpl(Functor&& f) : m_functor(std::move(f)) {}
+
+  void capture(const Kokkos::Cuda& exec, cudaGraph_t& graph) {
+    // Set the underlying stream to capture mode.
+    KOKKOS_IMPL_CUDA_SAFE_CALL(cudaStreamBeginCapture(
+        exec.cuda_stream(), cudaStreamCaptureModeGlobal));
```

Comment:
We may need to think which exec/stream to use for the capture. Because we may need to avoid "side effects" where the user may be doing something else on the same exec/stream that might get captured too.